app/models.py

```diff
@@ -0,0 +1,244 @@
+        """Verify token."""
+        s = Serializer(current_app.config['SECRET_KEY'])
+        try:
+            data = s.loads(token)
+        except SignatureExpired:
+            return None  # valid token, but expired
+        except BadSignature:
+            return None  # invalid token
+        user = User.query.get(data['id'])
+        return user
+
+    @login_manager.user_loader
+    def load_user(user_id):
+        """Return a user object if available or None otherwise."""
+        return User.query.get(int(user_id))
+
+    def __repr__(self):
+        """Represent objects with a readable format."""
+        return "<User(email='%s')>" % self.email
+
+
+class AnonymousUser(AnonymousUserMixin):
+    """Create an User with with AnonymousUserMixin properties."""
+
+    @staticmethod
+    def create_anonymous_user():
+        """Create anonymous user."""
+        anonymous_user = User(first_name='AnonymousUser',
+                              last_name='AnonymousUser', email='AnonymousUser')
+        anonymous_user.save()
+        return anonymous_user
+
+
+class ShortUrl(db.Model):
+    """Map ShortUrl class to short_url table in the database."""
+
+    __tablename__ = 'short_url'
+
+    id = db.Column(db.Integer, primary_key=True)
+    short_url = db.Column(db.String, unique=True)
+    is_active = db.Column(db.Boolean, default=True)
+    no_of_visits = db.Column(db.Integer, default=0)
+    deleted = db.Column(db.Boolean(), default=False, index=True)
+    when = db.Column(db.DateTime, default=db.func.current_timestamp(),
+                     nullable=False)
+    user_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=False)
+    user = db.relationship("User", back_populates="short_urls")
+    long_url_id = db.Column(db.Integer, db.ForeignKey('long_url.id'),
+                            nullable=False)
+    long_url = db.relationship("LongUrl", back_populates="short_urls")
+    logs = db.relationship("UrlActivityLogs", back_populates="short_url")
+
+    def change_long_url(self, long_url, user):
+        """Change the target url of a short_url and commit to database."""
+        new_long_url = LongUrl.query.filter_by(long_url=long_url).first()
+        old_long_url = self.long_url
+        if new_long_url in user.long_urls:
+            short_url = [short_url for short_url in user.short_urls if
+                         short_url.long_url_id == new_long_url.id][0]
+            return short_url
+        elif not new_long_url:
+            new_long_url = LongUrl(long_url=long_url)
+            db.session.add(new_long_url)
+            db.session.commit()
+            old_long_url.users.remove(user)
+            self.long_url = new_long_url
+            new_long_url.users.append(user)
+        if not old_long_url.users:
+            db.session.delete(old_long_url)
+        self.long_url = new_long_url
+        new_long_url.users.append(user)
+        self.when = db.func.current_timestamp()
+        self.no_of_visits = 0
```

Comment:
I am trying to reset it because I believe both short_url and long_url point to the same resource. So if long_url has been visited `n` times, so should short_url and if the user changes the long_url of a particular short_url and I don't reset it, that truth won't be reflected.